runtime/file.cpp

```diff
@@ -22,6 +28,35 @@ void OpenFile::set_path(OwningPtr<char> &&path, std::size_t bytes) {
   pathLength_ = bytes;
 }
 
+static int openfile_mkstemp(IoErrorHandler &handler) {
+#ifdef _WIN32
+  const unsigned int uUnique{0};
+  char tempDirName[MAX_PATH-14];
+  char tempFileName[MAX_PATH];
+  int nBufferLength{MAX_PATH-14};
+  nBufferLength = ::GetTempPathA(nBufferLength, tempDirName);
+  if (nBufferLength > MAX_PATH-14) {
+    return -1;
+  }
+  if (::GetTempFileNameA(tempDirName, "f18", uUnique, tempFileName) == 0) {
+    return -1;
+  }
+  int fd{::_open(tempFileName, _O_CREAT | _O_TEMPORARY, _S_IREAD | _S_IWRITE)};
+  if (fd < 0) {
```

Comment:
This part of the code is identical to the code in the other branch and can be moved after the `endif`.

Reply:
There's a unlink call in the non-windows code path which make it ugly. I pushed the change anyways.